src-tauri/src/telemetry_manager.rs

```diff
@@ -0,0 +1,321 @@
+    }
+
+    pub async fn initialize(
+        &mut self,
+        airdrop_access_token: Arc<RwLock<Option<String>>>,
+    ) -> Result<()> {
+        info!("Starting telemetry manager");
+        let telemetry_collection_enabled = self.config.read().await.telemetry_collection;
+        self.is_collecting_telemetry = telemetry_collection_enabled;
+        let _ = self
+            .start_telemetry_process(Duration::from_secs(60), airdrop_access_token)
+            .await;
+        Ok(())
+    }
+
+    async fn start_telemetry_process(
+        &mut self,
+        timeout: Duration,
+        airdrop_access_token: Arc<RwLock<Option<String>>>,
+    ) -> Result<(), TelemetryManagerError> {
+        let node_manager = self.node_manager.clone();
+        let cpu_miner = self.cpu_miner.clone();
+        let gpu_miner = self.gpu_miner.clone();
+        let config = self.config.clone();
+        let cancellation_token: CancellationToken = self.cancellation_token.clone();
+        let network = self.node_network;
+        let config_cloned = self.config.clone();
+
+        tokio::spawn(async move {
+            tokio::select! {
+                _ = async {
+                    info!("TelemetryManager::start_telemetry_process has  been started");
+                    loop {
+                        let telemetry_collection_enabled = config_cloned.read().await.telemetry_collection;
+
+                        let airdrop_access_token_internal = airdrop_access_token.read().await.clone();
+                        let airdrop_access_token_validated = airdrop_access_token_internal.clone().and_then(|t| {
+                            let key = DecodingKey::from_secret(&[]);
+                            let mut validation = Validation::new(Algorithm::HS256);
+                            validation.insecure_disable_signature_validation();
+
+                            let claims = match decode::<AirdropAccessToken>(&t, &key, &validation) {
+                                Ok(data) => Some(data.claims),
+                                Err(e) => {
+                                    warn!("Error decoding access token: {:?}", e);
+                                    None
+                                }
+                            };
+
+                            let now = std::time::SystemTime::now();
+                            let now_unix = now.duration_since(std::time::UNIX_EPOCH).unwrap_or_default().as_secs();
+
+                            if let Some(claims) = claims {
+                                if claims.exp < now_unix {
+                                    warn!("Access token has expired");
+                                    None
+                                } else {
+                                    Some(t)
+                                }
+                            } else {
+                                None
+                            }
+                        });
+
+                        if telemetry_collection_enabled {
```

Comment:
nit: perhaps do this before checking the JWT above, otherwise, every loop it will decode the jwt

Reply:
good catch